src/WrapperApp/components/InputEditor/InputEditorPanel.tsx

```diff
@@ -11,41 +11,48 @@ import { DEMO_MODE } from '../../../config/Config';
 import { InputFilesEditor } from './InputFilesEditor';
 import { readFile } from '../../../services/DataLoaderService';
 import { DragDropFiles } from './DragDropFiles';
-import { SimulationInputFiles, _defaultInputFiles } from '../../../types/ResponseTypes';
+import { ResponsePostJob, ResponseShConvert, ResponseTopasConvert, SimulationInputFiles, _defaultFlukaInputFiles, _defaultShInputFiles, _defaultTopasInputFiles } from '../../../types/ResponseTypes';
+import { RequestPostJob, RequestShConvert, RequestTopasConvert, SimulatorType } from '../../../types/RequestTypes';
 interface InputEditorPanelProps {
-	goToRun?: (InputFiles?: SimulationInputFiles) => void;
+	goToRun?: (simulator:SimulatorType, InputFiles?: SimulationInputFiles) => void;
 }
 
 type GeneratorLocation = 'local' | 'remote';
 
+type ConvertToInputFiles = 
+  | ((...args: RequestShConvert) => Promise<ResponseShConvert>)
+  | ((...args: RequestTopasConvert) => Promise<ResponseTopasConvert>);
+
 export default function InputEditorPanel({ goToRun }: InputEditorPanelProps) {
 	const { enqueueSnackbar } = useSnackbar();
 	const { editorRef } = useStore();
 	const { convertToInputFiles } = useShSimulation();
 	const { isConverterReady, convertJSON } = usePythonConverter();
 
 	const [isInProgress, setInProgress] = useState(false);
-	const [inputFiles, setInputFiles] = useState<SimulationInputFiles>(_defaultInputFiles);
+	const [inputFiles, setInputFiles] = useState<SimulationInputFiles>(_defaultShInputFiles);
 	const [generator, setGenerator] = useState<GeneratorLocation>('local');
+	const [simulator, setSimulator] = useState<SimulatorType>(SimulatorType.SHIELDHIT);
 
 	const [controller] = useState(new AbortController());
 
 	const handleConvert = useCallback(
 		async (editorJSON: EditorJson) => {
 			switch (generator) {
 				case 'remote':
-					return convertToInputFiles(editorJSON, controller.signal).then(res => {
+					convertToInputFiles(editorJSON, controller.signal).then(res => {
```

Comment:
Are you sure about this change?
it now can return undefined in remote switch case
![image](https://github.com/yaptide/ui/assets/26445545/49a3d243-dc3a-4d25-8bc4-4b38509d1d99)